pkg/cmd/cli/cmd/set/env.go

```diff
@@ -119,6 +128,9 @@ func RunEnv(f *clientcmd.Factory, in io.Reader, out io.Writer, cmd *cobra.Comman
 	//overwrite := kcmdutil.GetFlagBool(cmd, "overwrite")
 	resourceVersion := kcmdutil.GetFlagString(cmd, "resource-version")
 	outputFormat := kcmdutil.GetFlagString(cmd, "output")
+	secretName := kcmdutil.GetFlagString(cmd, "from-secret")
```

Comment:
Have you given though to how I would select a single data key from the secret?  Not required for an initial pull, but seems like a natural extension.